### PR TITLE
Downgrade to Java 8

### PR DESCRIPTION
**Pull request by KCodeYT**
This pull request includes:
- Direct downgrade from master branch without the usage of the old code from the broken `java-8` branch.

### CODE REVIEW

File: src/main/java/dev/waterdog/utils/ProxyConfig.java [hunk 1]
```diff
@@ -64,7 +64,7 @@
 
     @Path("permissions")
     @Comment("Case-Sensitive permission list for players")
-    private Object2ObjectOpenHashMap<String, List<String>> playerPermissions = new Object2ObjectOpenHashMap<>() {{
+    private Object2ObjectOpenHashMap<String, List<String>> playerPermissions = new Object2ObjectOpenHashMap<String, List<String>>() {{
```
Comment (KCodeYT) [inline src/main/java/dev/waterdog/utils/ProxyConfig.java:67]: This change was made so Java 8 knows which variable types is used in the inner class.

File: src/main/java/dev/waterdog/packs/types/ZipResourcePack.java [hunk 2]
```diff
@@ -54,8 +55,10 @@ public InputStream getStream(Path path) throws IOException {
 
     @Override
     public void saveToCache() throws IOException {
-        InputStream inputStream = Files.newInputStream(this.packPath);
-        this.cachedPack = inputStream.readAllBytes();
+        try (InputStream inputStream = Files.newInputStream(this.packPath)) {
+            byte[] allBytes = new byte[inputStream.available()];
+            this.cachedPack = Arrays.copyOf(allBytes, inputStream.read(allBytes));
+        }
```
Comment (KCodeYT) [inline src/main/java/dev/waterdog/packs/types/ZipResourcePack.java:61]: This code reads all the available bytes from the inputstream the same result as Java 11´s readAllBytes

**Comment by Alemiz112** [comment]
It might be easier to create new `java-8` branch from latest master.

**Comment by KCodeYT** [comment]
> It might be easier to create new `java-8` branch from latest master.

Then do it??

**Comment by KCodeYT** [comment]
> It might be easier to create new `java-8` branch from latest master.

Only those with write access to this repository might create new branches.

**Comment by Alemiz112** [comment]
Sure I can do it later.

**Comment by TobiasG-DE** [comment]
This PR can then be rebased on that java-8 branch and merged into that

**Comment by Alemiz112** [comment]
Force pushed latest changes from master to java-8 branch.

**Pull request closed without merging**